dipy/workflows/viz.py

```diff
@@ -150,6 +158,36 @@ def run(self, input_files, cluster=False, cluster_thr=15.,
                     print('Peak_dirs shape')
                     print(pam.peak_dirs.shape)
 
+            if ends(".npy"):
+
+                data = np.load(fname)
+                numpy_files.append(data)
```

Comment:
I am not sure how safe is appending numpy arrays into lists. I think you should use ``ArraySequence`` if you can. That is much preferred method. Remember that we had memory issues when adding arrays to lists. You can check if this problem persists by trying to delete this memory and see if the memory (list of arrays) is actually being removed from the RAM while the object is running.

Reply:
These files are very small in size. Eg: 100 integers only in one file. Mostly user will send one .npy file with pvalues in it (100 integers in the file). This should not cause any memory issues.